evm-adapters/src/sputnik/cheatcodes/cheatcode_handler.rs

```diff
@@ -1132,6 +1181,48 @@ impl<'a, 'b, B: Backend, P: PrecompileSet> Handler for CheatcodeStackExecutor<'a
                 }
             }
 
+            if input.len() >= 4 {
```

Comment:
```rust
// expected_calls would be `BTreeMap<Address, Vec<Vec<u8>>`
if let Some(expecteds) = self.state_mut().expected_calls(&code_address) {
    if let Some(found_match) = expecteds.iter().find(|expected| expected == input[..expected.len()]) {
        expecteds.remove(found_match);
    }
}
```

similarly for mocks:
```rust
// mock_calls would be `BTreeMap<Address, BTreeMap<Vec<u8>, Vec<u8>>`
if let Some(mocks) = self.state_mut().mock_calls(&code_address) {
    if let Some(mock_retdata) = mocks.get(&input) {
        return Capture::Exit((
                ExitReason::Succeed(ExitSucceed::Returned),
                ret.clone(),
        ))
    } else {
        if let Some(index) = mocks.iter().find(|mock| mock == input[..mock.len()]) {
               return Capture::Exit((
                    ExitReason::Succeed(ExitSucceed::Returned),
                    ret.clone(),
               ))
        }
    }
}
```

this is approximately what i mean - this code isnt perfect

Reply:
shud be ok now let me know what u think

<img src="https://media1.giphy.com/media/3o7abKhOpu0NwenH3O/giphy.gif"/>